src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -0,0 +1,148 @@
+        c.argument('top', options_list=['--top', '-t'], type=int, help='Maximum number of items to return. Default to 100.')
+        c.argument('all_', options_list=['--all'], action='store_true', help="List all items.")
+
+    with self.argument_context('appconfig kv lock') as c:
+        c.argument('key', required=True, help='Key to be locked.')
+        c.argument('label', help="If no label specified, lock entry with null label. Does NOT support filters like other commands")
+        c.argument('yes', options_list=['--yes', '-y'], action='store_true', help="Do not prompt for confirmation")
+
+    with self.argument_context('appconfig kv unlock') as c:
+        c.argument('key', required=True, help='Key to be unlocked.')
+        c.argument('label', help="If no label specified, unlock entry with null label. Does NOT support filters like other commands")
+        c.argument('yes', options_list=['--yes', '-y'], action='store_true', help="Do not prompt for confirmation.")
+
+    with self.argument_context('appconfig revision') as c:
+        c.argument('name', options_list=['--name', '-n'], help='The name of the App Configuration. You can configure the default name using `az configure --defaults app_configuration_store=<name>`', configured_default='app_configuration_store')
+        c.argument('connection_string', validator=validate_connection_string, configured_default='appconfig_connection_string',
+                   help="Connections of access key and endpoint of the App Configuration. Can be found using 'az appconfig credential list'. Users can preset it using `az configure --defaults appconfig_connection_string=<connection_string>` or environment variable with the name AZURE_APPCONFIG_CONNECTION_STRING.")
+
+    with self.argument_context('appconfig revision list') as c:
+        c.argument('key', help='If no key specified, return all keys by default. Support star sign as filters, for instance abc* means keys with abc as prefix. Similarly, *abc and *abc* are also supported.')
+        c.argument('label', help="If no label specified, list all labels. Support star sign as filters, for instance abc* means labels with abc as prefix. Similarly, *abc and *abc* are also supported.")
+        c.argument('datetime', validator=validate_datetime, help='Datetime to show history. Format: "YYYY-MM-DDThh:mm:ssZ". If no time zone specified, use UTC by default.')
+        c.argument('fields', nargs='+', help='Customize output fields.', validator=validate_query_fields, arg_type=get_enum_type(['key', 'value', 'label', 'content_type', 'etag', 'locked', 'last_modified']))
+        c.argument('top', options_list=['--top', '-t'], type=int, help='Maximum number of items to return. Default to 100.')
+        c.argument('all_', options_list=['--all'], action='store_true', help="List all revisions.")
```

Comment:
There is a **lot** of unnecessary duplication here. You should use appropriate scopes for your argument contexts to minimize this and/or group metadata into CLICommandArgument classes to reuse definitions.

Reply:
Done.